### ISSUE TITLE
Deadlock test fails with ArangoDB 3.2

### ISSUE THREAD

**Issue by frankmayer** [arangodb/arangodb-php#229]
@jsteemann there is another error in the test suite, which has to do with deadlock checking and handling.
It's a bit late in the night for me to look into this now, but I think it's also some change in ArangoDB itself. However, I wanted to post this here, just in case the tests of the client are off.

**Comment by jsteemann** [comment]
There are two reasons for this issue popping up:
- first, a last-minute config option was introduced for ArangoDB that allows controlling the execution of JavaScript code via HTTP calls. The default value of this option is to disallow this (which makes sense from my point of view), but the test that now fails relies on the old behavior (execution allowed). I have therefore locally adjusted the test code so it skips the test if the server is started with the "wrong" settings. Additionally I have adjusted the travis start script so the "old" option value is used to start the server.
- second, the test is only meaningful with the MMFiles engine. I have also modified the test so it gets skipped with storage engines other than MMFiles.

These two changes should fix the problem. I cannot push the changes to the PHP driver repository right now, because I am still waiting for some PR in the ArangoDB main repository to get accepted. When it gets accepted and there is a nightly build that contains the changes, I will push to the driver repository.
